data-processor/src/test/groovy/io/micronaut/data/model/jpa/criteria/CriteriaSpec.groovy

```diff
@@ -92,7 +92,7 @@ class CriteriaSpec extends AbstractCriteriaSpec {
             String query = getSqlQuery(criteriaQuery)
 
         expect:
-            query == '''SELECT book_."id",book_."author_id",book_."title",book_."total_pages",book_."last_updated" FROM "book" book_ WHERE (book_."id" IN (SELECT book_book_."id" FROM "book" book_book_ WHERE (book_book_."id" = 123)))'''
```

Comment:
Does not return field from the association if not requested by join for retrieval